src/asciidoc/docs/guides-general/build-custom-controls.adoc

```diff
@@ -0,0 +1,210 @@
+import './App.css';
+
+// Create an instance of CustomVcc.
+const myCustomCtrl = new CustomVcc();  // <2>
+
+function App() {
+
+  // A place to store the value entered.
+  let myStr = '';
+
+  // What to do after the user is done.
+  const finish = async () => {
+    // When a user clicks Save, update the value of the custom control, and save the file.
+    await myCustomCtrl.change(myStr);  // <5>
+    await myCustomCtrl.save();  // <6>
+    // alert( "Finished!" );
+  };
+
+  /*
+  // (Optional) Listen for changes to the consumer/parent app theme and match it in your custom control.
+  const [theme, setTheme] = ...;
+  useEffect(() => {
+    myCustomCtrl.onTheme((theme) => {  // <3>
+      setTheme(theme);
+    });
+  }, []);
+  */
+
+  // Listen for changes to the custom control's current state.
+  useState(() => {
+    myCustomCtrl.onUpdate((myProps) => {  // <3>
+      // alert( "myProps.value is " + myProps.value );  // <7>
+    });
+  });
+
+  // When the custom control has loaded, register it to trigger events from the consumer app.
+  useEffect(() => {
+    myCustomCtrl.register();  // <4>
+  }, []);
+
+  // Call the ready method once (with an empty dependency array)
+  // to let the platform know we have our listeners set up and
+  // to unblock the rendering and lookups for the rest of our application.
+  useEffect(() => {
+    Koji.ready();
+  }, []);
+
+  return (
+    <div className="App">
+      <header className="App-header">
+        <h1>My Custom Control</h1>
+        <p>Enter any text:</p>
+        <input
+          onChange={(e) => {
+            myStr = e.currentTarget.value;
+          }}
+        />
+        <button onClick={finish}>{'Save'}</button>
+      </header>
+    </div>
+  );
+}
+
+export default App;
+
+----
+
+[.callout1]#Imports the `CustomVcc` class from the @withkoji/custom-vcc-sdk package.# +
+[.callout2]#Creates an instance of `CustomVcc`.# +
+[.callout3]#Starts event listeners to listen for updates.# +
+[.callout4]#After the customer control loads, registers it to trigger events from the parent editor.# +
+[.callout5]#When a user changes the value, updates the custom control.# +
+[.callout6]#Saves the file.# +
+[.callout7]#For testing. Displays the value returned by the custom control.#
+
+To customize the forked code for your custom control:
+
+. Modify any of the default settings and parameters.
+. Add code required by your custom control, such as:
++
+* Elements in the `return` section that users can interact with.
+* Commands in the `finish` section to perform additional processing on the captured input.
+. Clean up unneeded code.
+. When done, publish your new custom control app.
+
+Learn more: <<customvcc-reference#>>
+
+==== Provide a link to your custom control
+
+Any app that uses your custom control must call `Koji.ui.capture.custom` and provide a `koji-vccs.com` subdomain or URL that points to the custom control.
+
+* You can determine the assigned URL of the custom control.
```

Comment:
Suggest removing. If we recommend publishing on a subdomain of koji-vcc.com, let's focus on those instructions rather than providing every option in this tutorial.

Reply:
Fixed.